Do not scale WidthHeuristics when max_width less than 100

tspiteri opened the pull request:
Fixes #2644.

nrc commented on src/config/options.rs:
```diff
@@ -241,8 +241,14 @@ impl WidthHeuristics {
-        let mut max_width_ratio: f32 = max_width as f32 / 100.0; // 100 is the default width -> default ratio is 1
-        max_width_ratio = (max_width_ratio * 10.0).round() / 10.0; // round to the closest 0.1
+        // 100 is the default width -> default ratio is 1
+        let max_width_ratio = if max_width > 100 {
```
COuld we pull the default max width out into a const?

tspiteri commented:
I added the `DEFAULT_MAX_WIDTH` constant in the function, added a test for the issue in commit 2, and added a test to check that width heuristics work for larger widths in commit 3. In commit 3 I also had to touch *config_type.rs* to make sure `self.set_heuristics()` is called if `max_width` changes, otherwise `WidthHeuristics` are not updated properly for the test's `rustfmt-max_width`.

nrc commented:
Thank you, looks great!
